src/Synthesiser.cpp

```diff
@@ -289,28 +289,47 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             PRINT_END_COMMENT(out);
         }
 
-        void visitQuery(const RamQuery& insert, std::ostream& out) override {
+        void visitQuery(const RamQuery& query, std::ostream& out) override {
```

Comment:
This method is quite long now. Can it be split into helper methods? e.g. the actual scan part could be in a helper, as could the check for empty input relations.

Reply:
Yes - in the next pull request. The issue is that we generate C++ conditions that should be expressed in RAM rather than C++. The interpreter will benefit from these changes as well.
Also that we directly go after scans/search etc is quite ugly. We need to refactor this method.